src/wasm/cimbar_js/CMakeLists.txt

```diff
@@ -28,13 +28,43 @@ target_link_libraries(cimbar_js
 	${OPENCV_LIBS}
 )
 
-if(DEFINED USE_WASM)
-set_target_properties( cimbar_js PROPERTIES LINK_FLAGS "-s DEMANGLE_SUPPORT=1 --bind -s SINGLE_FILE=1 -Os -s FULL_ES2=1 -s USE_WEBGL2=1 -s MIN_WEBGL_VERSION=2 -s MAX_WEBGL_VERSION=2 -s USE_GLFW=3 -s FILESYSTEM=0 -s TOTAL_MEMORY=134217728 -s EXPORTED_FUNCTIONS='[\"_render\", \"_initialize_GL\", \"_encode\"]'" )
+### various emscripten shenanigans follow
+
+# asm.js vs wasm
+if(USE_WASM EQUAL "2")
+	set (LINK_WASM_OPT "-s SINGLE_FILE=1")
+else()
+	set (LINK_WASM_OPT "-s WASM=1")
+endif()
 
-install(
-	TARGETS cimbar_js
-	DESTINATION ${libcimbar_SOURCE_DIR}/web
+# extra link flags for emscripten
+set (LINK_WASM_LIST
+	--bind -Os
+	${LINK_WASM_OPT}
```

Comment:
Took some minor cmake jank to get this to work, but I think this is much more readable than it was...